config/package_resource.go

```diff
@@ -100,6 +110,78 @@ func (p *packageResouce) validateFile(file *agentendpointpb.OSPolicy_Resource_Fi
 	return nil
 }
 
+type packageInfoCache map[string]packageInfo
+
+type packageInfo struct {
+	PkgInfo    *packages.PkgInfo
+	LastLookup time.Time
+}
+
+func getPackageInfoCacheKey(pkgFile *agentendpointpb.OSPolicy_Resource_File) (string, error) {
+	// We use the base64 encoded binary proto as the key.
+	raw, err := proto.Marshal(pkgFile)
+	if err != nil {
+		return "", err
+	}
+	return base64.RawStdEncoding.EncodeToString(raw), nil
```

Comment:
Would it be better to compute the hash of the contents of the file to reduce the key size? A large file could lead to a very large key otherwise.

Reply:
We are not caching the file, just the proto field File for a package resource. This is always very small and most likely much smaller than a hash would be (sha256 hash is 64 in hex format)